### PR TITLE
Implemented VSX splat instructions

### PR DESCRIPTION
**Pull request by billionai**
Moved the 2 VSX intructions that were already implemented to using decodetree, and implemented the 3 new splat instructions, along with the macro REQUIRE_VSX.

I'm not sure why xxspltib might require altivec instead of VSX... any of you know why  that may be, and if that should be implemented in the other xxsplt* instructions? xxspltw didn't have it either....

### CODE REVIEW

File: target/ppc/insn64.decode [hunk 1]
```diff
@@ -24,6 +24,14 @@
                 ...... rt:5 ra:5 ................       \
                 &PLS_D si=%pls_si
 
+# Format 8RR:D
+%8rr_si         32:s16 0:16
+%8rr_xt         16:1 21:5
+&8RR_IX         xt ix si:int32_t
+@8RR_IX         ...... .. .... .. .. ................ \
```
[the remark this answers is not in the document]
Comment (billionai) [reply target/ppc/insn64.decode:31]: Oh, right, you mentioned that and I forgot to edit it b4 the PR.

File: target/ppc/insn32.decode [hunk 2]
```diff
@@ -43,6 +43,10 @@
 &X_bfl          bf l:bool ra rb
 @X_bfl          ...... bf:3 - l:1 ra:5 rb:5 ..........- &X_bfl
 
+&X_vfs          xt imm:uint8_t
+%x_vfs_xt       0:1 21:5
+@X_vfs          ...... ..... .. imm:8 .......... .      &X_vfs xt=%x_vfs_xt
```
[the remark this answers is not in the document]
Comment (billionai) [reply target/ppc/insn32.decode:48]: I had a good reason, I think it was "Vector-Facility-Scalar" or something, but I can change the name for sure

File: target/ppc/translate/vsx-impl.c.inc [hunk 3]
```diff
@@ -2106,6 +2116,22 @@ static void gen_xvxsigdp(DisasContext *ctx)
     tcg_temp_free_i64(xbl);
 }
 
+static bool trans_XXSPLTIDP(DisasContext* ctx, arg_8RR_D *a)
+{
+    REQUIRE_VSX(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+
+    TCGv_i64 t0;
+    t0 = tcg_temp_new_i64();
```
[the remark this answers is not in the document]
Comment (billionai) [reply target/ppc/translate/vsx-impl.c.inc]: This was the first implementation, that was later changed to a single call to a different TCG function. This code slipped through my rebases

**Review by alqotel — APPROVED** [review]
LGTM

**Review by pherde — APPROVED** [review]
LGTM

**Pull request merged**